evalml/tests/automl_tests/test_automl.py

```diff
@@ -1763,29 +1768,29 @@ def test_automl_error_callback(mock_fit, mock_score, X_y_binary, caplog):
     X, y = X_y_binary
     msg = 'all your model are belong to us'
     mock_fit.side_effect = Exception(msg)
-    automl = AutoMLSearch(problem_type="binary", error_callback=None)
+    automl = AutoMLSearch(problem_type="binary", error_callback=None, train_best_pipeline=False)
```

Comment:
Why set this to false in this unit test? I thought this unit test was completely unrelated to the `best_pipeline` accessor

Reply:
The test mocks the `BinaryClassificationPipeline.fit` method, which means that when I call `best_pipeline.fit()`, it returns an exception that isn't captured and ends up breaking out of the code. This code just checks that callbacks within `_compute_cv_scores` works, so any `fit` call outside of that fails the test here.